src/electronic-id.cpp

```diff
@@ -112,7 +107,22 @@ const std::map<byte_vector, ElectronicIDConstructor> SUPPORTED_ATRS {
      constructor<ElectronicID::Type::CzeEID>},
 };
 
-inline std::string byteVectorToHexString(const byte_vector& bytes)
+// Holds ATR pattern, mask, and constructor for variable ATR cards.
+struct MaskedATREntry
+{
```

Comment:
```suggestion
{
    template <size_t PatSize, size_t MaskSize>
    constexpr MaskedATREntry(const byte_type (&_pat)[PatSize], const byte_type (&_mask)[MaskSize], ElectronicIDConstructor &&_constructor)
        : pattern(_pat, std::next(_pat, PatSize))
        , mask(_mask, std::next(_mask, MaskSize))
        , constructor(std::move(_constructor))
    {
        static_assert(PatSize == MaskSize, "Pattern and mask sizes must be equal.");
    }
```

Reply:
Great idea, I took it even further by using a single template parameter.